doc/manual/src/design/store/objects.md

```diff
@@ -0,0 +1,73 @@
+# Store Objects
+
+Data in Nix is chiefly organized into *store objects*.
+A store object is the pair of
+
+  - A (root) file system object
+  - A set of references to store objects
```

Comment:
```suggestion
  - a file system object
  - a set of references to store objects.
```

It's not clear what "root" would mean here, because we have not yet defined what a file system object is, and the root notion only explained at the end.